lfs/config.go

```diff
@@ -57,18 +57,28 @@ func ObjectUrl(endpoint Endpoint, oid string) (*url.URL, error) {
 	return u, nil
 }
 
-func (c *Configuration) Endpoint() Endpoint {
+func (c *Configuration) Endpoint() (endpoint Endpoint) {
+	defer func() {
+		if u, err := url.Parse(endpoint.Url); err == nil {
+			if u.User != nil {
+				fmt.Fprintln(os.Stderr, "warning: configured LFS endpoint contains credentials")
+			}
+		}
+	}()
```

Comment:
Using a `defer` here seems odd.  This isn't really cleaning up a resource.  I wonder if we should just have a `NewEndpoint()` function that encapsulates this?